### PR TITLE
Update bootstra-table-export.js and tableExport.js

### PR DESCRIPTION
**Pull request by dobobaie**
You can select hidden or not the colums.
Clean code.
Look at https://jsfiddle.net/v12zLf0a/5/

### CODE REVIEW

File: src/extensions/export/bootstrap-table-export.js [hunk 1]
```diff
@@ -87,11 +118,11 @@
                             that.$el.tableExport($.extend({}, that.options.exportOptions, {
                                 type: type,
                                 escape: false
-                            }));
+                            }), that.options.data);
                         };
 
                     if (that.options.exportDataType === 'all' && that.options.pagination) {
-                        that.$el.one(that.options.sidePagination === 'server' ? 'post-body.bs.table' : 'page-change.bs.table', function () {
+                        that.$el.one('load-success.bs.table page-change.bs.table', function () {
```
Comment (djhvscf) [inline src/extensions/export/bootstrap-table-export.js]: Are you sure that this change will work as expected?? This line of code was changed in order to fix an issue related to pagination and the _Export All_ option

Comment (dobobaie) [inline src/extensions/export/bootstrap-table-export.js]: Ohhh i hadn't seen that sorry !

**Comment by dobobaie** [comment]
@djhvscf i reviewed my code with pagination !
Demo: https://jsfiddle.net/v12zLf0a/6/
bootstrap-table-export: https://github.com/dobobaie/bootstrap-table/blob/develop/src/extensions/export/bootstrap-table-export.js
tableExport: https://github.com/dobobaie/tableExport.jquery.plugin/blob/master/tableExport.js

File: src/extensions/export/bootstrap-table-export.js [hunk 2]
```diff
@@ -8,12 +8,13 @@
     var sprintf = $.fn.bootstrapTable.utils.sprintf;
 
     var TYPE_NAME = {
+        //png: 'PNG',
+        //txt: 'TXT',
```
Comment (wenzhixin) [inline src/extensions/export/bootstrap-table-export.js]: Why do you disable the `png` and `txt` types?

File: src/extensions/export/bootstrap-table-export.js [hunk 3]
```diff
@@ -8,12 +8,13 @@
...
         json: 'JSON',
         xml: 'XML',
-        png: 'PNG',
         csv: 'CSV',
-        txt: 'TXT',
         sql: 'SQL',
+        html: 'HTML',
```
Comment (wenzhixin) [inline src/extensions/export/bootstrap-table-export.js]: https://github.com/hhurz/tableExport.jquery.plugin#export-html-table-to

File: src/extensions/export/bootstrap-table-export.js [hunk 4]
```diff
@@ -23,8 +24,7 @@
     $.extend($.fn.bootstrapTable.defaults, {
         showExport: false,
         exportDataType: 'basic', // basic, all, selected
-        // 'json', 'xml', 'png', 'csv', 'txt', 'sql', 'doc', 'excel', 'powerpoint', 'pdf'
-        exportTypes: ['json', 'xml', 'csv', 'txt', 'sql', 'excel'],
+        exportTypes: ['json', 'html', 'xml', 'csv', 'txt', 'sql', 'pdf', 'excel', 'doc', 'powerpoint'],
```
Comment (wenzhixin) [inline src/extensions/export/bootstrap-table-export.js]: I think the default exportTypes should be useful.

**Comment by wenzhixin** [comment]
Thanks for your @dobobaie , please submit the changes of `tableExport.js` to https://github.com/hhurz/tableExport.jquery.plugin/pulls.

**Comment by dobobaie** [comment]
@wenzhixin :+1:  :) 
New review https://jsfiddle.net/v12zLf0a/9/ we can export checked row.
File: https://cdn.rawgit.com/dobobaie/bootstrap-table/30f0729a326326463e384161f203c87221069931/src/extensions/export/bootstrap-table-export.js

File: src/extensions/export/bootstrap-table-export.js (hunk 4 above)
Comment (hhurz) [inline src/extensions/export/bootstrap-table-export.js]: @wenzhixin Thats on your decision. @dobobaie rewrote nearly the complete code of the tableExport.js plugin. There are too many changes (325 additions, 1,273 deletions) that prevents me to assess that work and I'm not able to merge dobobaie's pull request with my code base. I suggest that dobobaie continuous his work in his own fork or even shifts it into a complete new project with its own name to become more distinguishable from tableExport.js. dobobaie describes the changes he made with "more light, more clean and easy to dev ext." It's out of my scope to decide if dobobaie's version is suitable to replace my version of the plugin in bootstrap-table. 

Note: This comment should never be a disqualification of dobobaie's work!

Comment (dobobaie) [inline src/extensions/export/bootstrap-table-export.js]: @hhurz thank you again for your notes I apperciate very much. While wenzhixin take a decision, i continue in my own fork (must to add txt and png export and some options).

For the difference between actual export options and my export options, i can write a "convert options" while users edit their code.

**Comment by dobobaie** [comment]
I've new version https://jsfiddle.net/v12zLf0a/10/
txt and png it's ok (i will finish pdf and png later)
bootstrap-table-export => https://github.com/dobobaie/bootstrap-table/blob/8c0471902d7cd9bb627fceb2752925a3ced32903/src/extensions/export/bootstrap-table-export.js
tableExport => https://github.com/dobobaie/tableExport.jquery.plugin/blob/48cdcc3e279de97cbbdebfe48f45564990c41530/tableExport.js

**Comment by wenzhixin** [comment]
Sorry we cannot receive this PR, thanks anyway.

**Pull request closed without merging**